test/tests/functional/pbs_strict_ordering.py

```diff
@@ -164,3 +163,104 @@ def test_t2(self):
                            id=j5id,
                            max_attempts=30,
                            interval=2)
+
+    def test_zero_backfill_depth_on_queue(self):
+        """
+        Test if scheduler tries to run a job when strict ordering is enabled
+        and backfill_depth is set to 0 on the queue
```

Comment:
These tests are fine, but I think we can make them slightly better.  Instead of having 1 cpu and submitting 3 identical jobs, how about we have 2 cpus and submit jobs requesting 1 cpu, 2 cpus, 1cpu.  If backfill_depth wasn't 0, we'd run jobs 1 and 3.  Since backfill_depth is 0, we won't run job 3.